docs/contribution-guide/how-to-contribute.md

```diff
@@ -0,0 +1,280 @@
+
+!!! example "@reviewer: see bug [report template](https://github.com/ELS-RD/kernl/blob/feat/contribution-guide-doc/.github/ISSUE_TEMPLATE/bug.yml)"
+
+#### Feature requests
+
+If you would like to **request a new feature** or **an enhancement**, 
+but are not planning to open a Pull Request, you can **[open a dedicated issue](todo)**.
+You can also use the **[discussion forum](todo)** for feature requests that you would like to discuss.
+
+!!! example "@reviewer: see [feature request template](https://github.com/ELS-RD/kernl/blob/feat/contribution-guide-doc/.github/ISSUE_TEMPLATE/feature.yml)"
+
+#### Proposals
+
+If you intend to **make non-trivial changes to existing implementations**, we recommend that you **file an Issue** with the **[proposal template](todo)**. This allows us to reach agreement on your proposal before you put any effort into it.
+
+!!! example "@reviewer: see [proposal template](https://github.com/ELS-RD/kernl/blob/feat/contribution-guide-doc/.github/ISSUE_TEMPLATE/proposal.yml)"
+
+#### Documentation
+
+If you would like to **add or improve documentation** or **tutorials**,
+but are not planning to open a Pull Request, you can **[open a dedicated issue](todo)**.
+You can also use the **[discussion forum](todo)** to discuss this topic.
+
+!!! example "@reviewer: see [documentation template](https://github.com/ELS-RD/kernl/blob/feat/contribution-guide-doc/.github/ISSUE_TEMPLATE/documentation.yml)"
+
+## Development
+
+### Branch Organization
+
+Kernl works with a `main` branch. This is the source of truth.
+
+Each **new Pull Request**, for a new feature, bug fix, or other, must be done on **a new branch**.
+
+#### Standardization of branch names
+
+The name of a new branch must respect the following format:
+```
+<type>(scope)/<subject>
+``` 
+
+??? question "Which types and characters are allowed?"
+
+    The types of new branches are inspired by the values of **[Conventional commit](https://www.conventionalcommits.org/en/v1.0.0/)**.
+    
+    **List of possible types:**
+
+    - `feat`: new functionality, optimization or implementation of a proposal.
+    - `fix`: bug fix.
+    - `docs`: ajout ou modification de la documentation.
+    - `refactor`: a change in the code that does not result in any difference in behavior.
+    - `test`: adding tests, refactoring tests. No production code change.
+    - `chore`: upgrading dependencies, releasing new versions. Tasks that are regularly done for maintenance purposes.
+    - `misc`: anything else that doesn't change production code, yet is not test or chore. e.g. updating GitHub actions workflow.
+
+    **Allowed characters for the subject:** 
+        ```regexp
+        [a-z_-]
+        ```
+    !!! example "Examples"
+        - feat/debugger
+        - feat/backward_layernorm
+        - refactor/refactor-kernels
+        - docs/contribution-guide-doc
+
+### Installation
```

Comment:
should we put this in two different places ? we can forgot to update

Reply:
The entire installation? I agree, I think this kind of information should be centralized in one place. Currently it is in the README and not in the doc. We can refer to the README here.